tests/js/sentry-test/reactTestingLibrary.tsx

```diff
@@ -116,38 +127,74 @@ function makeAllTheProviders(options: ProviderOptions) {
       getCurrentLocation: jest.fn(() => ({pathname: '', query: {}})),
     });
 
-    // By default react-router 6 catches exceptions and displays the stack
-    // trace. For tests we want them to bubble out
-    function ErrorBoundary(): React.ReactNode {
-      throw useRouteError();
-    }
-
-    const routes: RouteObject[] = [
-      {
-        path: '*',
-        element: wrappedContent,
-        errorElement: <ErrorBoundary />,
-      },
-    ];
-
-    const memoryRouter = createRouter({
-      future: {v7_prependBasename: true},
-      history,
-      routes,
-    }).initialize();
-
     return (
       <CacheProvider value={{...cache, compat: true}}>
         <ThemeProvider theme={lightTheme}>
           <QueryClientProvider client={makeTestQueryClient()}>
-            <RouterProvider router={memoryRouter} />
+            {wrappedContent}
           </QueryClientProvider>
         </ThemeProvider>
       </CacheProvider>
     );
   };
 }
 
+function makeRouter({
+  children,
+  history,
+}: {
+  history: MemoryHistory;
+  children?: React.ReactNode;
+}) {
+  // By default react-router 6 catches exceptions and displays the stack
+  // trace. For tests we want them to bubble out
+  function ErrorBoundary(): React.ReactNode {
+    throw useRouteError();
+  }
+
+  const routes: RouteObject[] = [
+    {
+      path: '*',
+      element: children,
+      errorElement: <ErrorBoundary />,
+    },
+  ];
+
+  const router = createRouter({
+    future: {v7_prependBasename: true},
+    history,
+    routes,
+  }).initialize();
+
+  return router;
+}
+
+class TestRouter {
```

Comment:
Instead of returning the router directly, created a utility class to make testing more foolproof. `router.location` will access `router.state.location` and `navigate` wraps the navigate function in an act.

Reply:
Yeah this seems good 👍